File: app/schema/inventory_transaction.py
from pydantic import BaseModel
from decimal import Decimal
from datetime import datetime
from typing import Optional
from app.enums import InventoryTransactionType, ApprovalStatus

class InventoryTransaction(BaseModel):
    transaction_num: int
    transaction_type: InventoryTransactionType
    quantity: Decimal
    transaction_date: datetime
    approved_by: Optional[str] = None
    approver_name: Optional[str] = None
    created_by: str
    creator_name: str
    approval_status: ApprovalStatus
    invoice_id: Optional[int] = None
    product_num: int
    price_per_unit: Decimal
    batch_num: Optional[int] = None
    reason: Optional[str] = None

File: frontend/src/types/inventoryTransaction.ts
import type { ApprovalStatus, InventoryTransactionType  } from "./enums"

export type InventoryTransaction = {
    transaction_num: number
    transaction_type: InventoryTransactionType
    quantity: string
    internal_units: string
    transaction_date: string
    approved_by: string | null
    approver_name: string | null
    created_by: string
    creator_name: string
    approval_status: ApprovalStatus
    invoice_id: number |null
    product_num: number
    price_per_unit: string
    batch_num: number | null
    reason: string | null
}